src/cmap/establish/handshake/mod.rs

```diff
@@ -229,6 +244,32 @@ impl Handshaker {
                 .map(|server_first| client_first.into_first_round(server_first))
         });
 
+        // Check that master reply has a compressor list and unpack it
+        if let Some(ref server_compressors) = is_master_reply.command_response.compressors {
+            // Check that client has a compressor list and unpack it
+            if let Some(ref client_compressors) = self.compressors {
+                // Use the Client's first compressor choice that the server supports
+                for client_compressor in client_compressors {
```

Comment:
I think this would be clearer using iterators, i.e. something like
```
  if let Some(compressor) = client_compressors.iter().find(|c| server_compressors.contains(c)) {
    conn.compressor = compressor.parse().ok();
    ...
  }
```